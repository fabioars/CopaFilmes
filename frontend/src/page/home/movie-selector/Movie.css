.Movie {
    position: relative;
    display: flex;
    align-items: flex-start;
    justify-content: center;
    flex-direction: column;
    background-color: #fff;
    color: var(--primary-color);
    height: 120px;
    cursor: pointer;
    padding: 10px 30px 10px 66px;
    box-sizing: border-box;
}

.Movie:hover {
    background-color: #ccc;
}

.Movie:before {
    content: '';
    display: block;
    width: 26px;
    height: 26px;
    position: absolute;
    left: 20px;
    top: 50%;
    border: 1px solid var(--primary-color);
    transform: translateY(-50%);
}

.Movie--selected {
    border: 2px solid var(--bg-color-darken);
}

.Movie:after {
    content: '';
    display: block;
    width: 18px;
    height: 18px;
    position: absolute;
    left: 25px;
    top: 50%;
    transform: translateY(-50%);
}

.Movie--selected.Movie:after {
    background-color: var(--primary-color);
}

.Movie__title {
    font-weight: bold;
}

.Movie__year {
    font-size: 14px;
}
