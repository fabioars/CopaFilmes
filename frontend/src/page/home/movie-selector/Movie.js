import React from 'react';
import './Movie.css';

export default function Movie({ movie, onClick, isSelected }) {
    const classNames = ['Movie']
    if(isSelected) {
        classNames.push('Movie--selected');
    }

    return (
        <div className={classNames.join(' ')} onClick={onClick}>
            <span className="Movie__title">{movie.titulo}</span>
            <span className="Movie__year">{movie.ano}</span>
        </div>
    );
};
