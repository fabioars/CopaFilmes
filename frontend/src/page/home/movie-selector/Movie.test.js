import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import Movie from './Movie';

describe('Movie', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });


    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    describe('Header', () => {
        it('renders correctly', () => {
            const tree = renderer.create(
                <Movie movie={{ id: 'tt3606756', titulo: 'Os Incríveis 2', ano: 2018, nota: 8.5 }} />
                ).toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should call onClick props onClick', () => {
            const callback = jest.fn();

            act(() => {
                render(
                    <Movie
                        movie={{ id: 'tt3606756', titulo: 'Os Incríveis 2', ano: 2018, nota: 8.5 }}
                        onClick={callback}
                    />,
                    container
                );
            });

            container.querySelector('.Movie').click();
            expect(callback.mock.calls.length).toBe(1);
        });

        it('should have a Movie--selected class if isSelected iquals true', () => {
            act(() => {
                render(
                    <Movie
                        movie={{ id: 'tt3606756', titulo: 'Os Incríveis 2', ano: 2018, nota: 8.5 }}
                        isSelected={true}
                    />,
                    container
                );
            });

            expect(document.querySelector('.Movie--selected')).toBeTruthy();
        });
    });

});
