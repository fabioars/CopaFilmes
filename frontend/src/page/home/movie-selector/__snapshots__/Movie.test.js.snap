// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Header renders correctly 1`] = `
<div
  className="Movie"
>
  <span
    className="Movie__title"
  >
    Os Incr√≠veis 2
  </span>
  <span
    className="Movie__year"
  >
    2018
  </span>
</div>
`;
